commandapi-platforms/commandapi-bukkit/commandapi-bukkit-test/commandapi-bukkit-test-tests/src/test/java/dev/jorel/commandapi/test/CommandNamespaceTests.java

```diff
@@ -0,0 +1,767 @@
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.mockito.Mockito;
+
+import java.lang.reflect.Method;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+
+/**
+ * Tests for testing if namespaces work correctly
+ */
+public class CommandNamespaceTests extends TestBase {
+
+	/*********
+	 * Setup *
+	 *********/
+
+	@BeforeEach
+	public void setUp() {
+		super.setUp();
+	}
+
+	@AfterEach
+	public void tearDown() {
+		super.tearDown();
+	}
+
+	Player enableWithNamespaces() {
+		// Register minecraft: namespace. MockBukkit doesn't do this on their own
+		// Simulate `CraftServer#setVanillaCommands`
+		MockPlatform<Object> mockPlatform = MockPlatform.getInstance();
+		SimpleCommandMap commandMap = mockPlatform.getSimpleCommandMap();
+		for (CommandNode<Object> node : mockPlatform.getBrigadierDispatcher().getRoot().getChildren()) {
+			commandMap.register("minecraft", mockPlatform.wrapToVanillaCommandWrapper(node));
+		}
+
+		// Run the CommandAPI's enable tasks, especially `fixNamespaces`
+		assertTrue(CommandAPI.canRegister()); // Make sure we weren't already enabled
+		disablePaperImplementations();
+		Bukkit.getPluginManager().callEvent(new ServerLoadEvent(ServerLoadEvent.LoadType.STARTUP));
+		assertDoesNotThrow(() -> server.getScheduler().performOneTick());
+		assertFalse(CommandAPI.canRegister());
+
+		// Get a CraftPlayer for running VanillaCommandWrapper commands
+		Player runCommandsPlayer = Mockito.mock(MockPlatform.getInstance().getCraftPlayerClass());
+		// Give player permission to run command
+		Mockito.when(runCommandsPlayer.hasPermission(any(String.class))).thenReturn(true);
+		// Get location is used when creating the BrigadierSource in MockNMS
+		Mockito.when(runCommandsPlayer.getLocation()).thenReturn(new Location(null, 0, 0, 0));
+
+		// Provide proper handle as VanillaCommandWrapper expects
+		Method getHandle = assertDoesNotThrow(() -> runCommandsPlayer.getClass().getDeclaredMethod("getHandle"));
+		Object brigadierSource = Brigadier.getBrigadierSourceFromCommandSender(runCommandsPlayer);
+		Object handle = Mockito.mock(getHandle.getReturnType(), invocation -> brigadierSource);
+		// This is a funny quirk of Mockito, but you don't need to put the method call you want to mock inside `when`
+		//  As long as the method is called, `when` knows what you are talking about
+		//  That means we can mock `CraftPlayer#getHandle` indirectly using reflection
+		//  See the additional response in https://stackoverflow.com/a/10131885
+		assertDoesNotThrow(() -> getHandle.invoke(runCommandsPlayer));
+		Mockito.when(null).thenReturn(handle);
+
+		return runCommandsPlayer;
+	}
+
+	/*********
+	 * Tests *
+	 *********/
+
+	@Disabled
```

Comment:
Please, go ahead. They do work in-game, I did verify that so that's nothing that should stop us from merging.
Figuring this out would be great though!